python/ray/experimental/channel/nccl_group.py

```diff
@@ -103,6 +110,16 @@ def __init__(
                 cuda_stream, device_id=device.index
             )
 
+            if dedicated_streams:
+                import torch
+
+                self._send_stream = cp.cuda.ExternalStream(
```

Comment:
what's the mem overhead of additional stream btw? (is there gpu mem overhead?)

Reply:
The stream itself should not have much memory overhead, although kernel launches, concurrent execution, and memory copies related to its operations could take extra memory. But this is the price to pay for faster execution, right? I think this boils down to the workload and a tradeoff analysis with profiling, but we probably want to enable overlapping with sufficient memory.

Were you suggesting any alternative way of doing things?

Reply:
Oh no. I just want to understand the exact memory tradeoff. I think it is the right thing to do,but it is important to understand how much impact it can cause.